crates/indexer/src/bigquery_client.rs

```
@@ -0,0 +1,150 @@
+            None,
+        )
+        .await
+        .expect("Create raw Bigquery Client successfully");
+        let data_set_path = format!("projects/{}/datasets/{}/tables/", project_id, dataset_name,);
+        Self {
+            client,
+            stream_map: Mutex::new(HashMap::new()),
+            data_set_path,
+        }
+    }
+    /// Returns the default stream for the data; if not present, create one.
+    async fn get_stream(&self, table: String) -> String {
+        match self.stream_map.lock().await.get(&table) {
+            Some(stream_id) => stream_id.to_owned(),
+            None => {
+                let write_stream_resp = self
+                    .client
+                    .get()
+                    .create_write_stream(tonic::Request::new(CreateWriteStreamRequest {
+                        parent: format!("{}{}", self.data_set_path, table),
+                        write_stream: Some(WriteStream {
+                            r#type: i32::from(write_stream::Type::Committed),
+                            ..WriteStream::default()
+                        }),
+                    }))
+                    .await
+                    .expect("Create stream successfully.");
+                write_stream_resp.into_inner().name
+            }
+        }
+    }
+
+    /// Sends the data to bigquery streaming api.
+    pub async fn send_data(&self, append_row_req: TypedAppendRowsRequest) -> anyhow::Result<()> {
+        let req = match append_row_req {
+            TypedAppendRowsRequest::Transactions(mut data) => {
+                data.write_stream = self.get_stream("transactions".to_string()).await;
+                data
+            }
+        };
+        match self
+            .client
+            .get()
+            .append_rows(tonic::Request::new(stream::iter(vec![req])))
+            .await
+        {
+            Ok(_) => Ok(()),
+            Err(_) => bail!("Failed to send data to BigQuery."),
```

can we not discard the error here, to help with debug?